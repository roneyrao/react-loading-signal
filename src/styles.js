export default from './styles/FlexLoading.css';
