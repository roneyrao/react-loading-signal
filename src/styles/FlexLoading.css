.loading {
	width:100%;
	top:50%;
	transform: translateY(-50%);
	z-index:999;
	visibility:hidden;
	text-align:center;
	word-wrap:normal;
	white-space:nowrap;
}
.masked{
	top:0;
	right:0;
	bottom:0;
	left:0;
	z-index:9999;
	display:flex;
	justify-content:center;
	align-items:center;
	transform: none;
}
.masked:before{
	content:'';
	top:0;
	right:0;
	bottom:0;
	left:0;
	background:rgba(150, 150, 150, 0.4);
	position: absolute;
}
.masked.loading>* {
	transform: none;
	z-index: 1;
}

.local{
	composes:loading;
	position:absolute;
}
.localMasked{
	composes:local masked;
}

.global{
	composes:loading;
	position:fixed;
}
.globalMasked{
	composes:global masked;
}

.messageList{
	margin-top:1em;
	line-height:2em;
}
/*theme 'blobs'*/
.blobs{
    width: 100%;
	height:1em;
	background:url(./images/blobs.gif) no-repeat center center;
	background-size:auto 100%;
}
/*theme 'spinning'*/
.spinning{
	min-width:3em;
    padding: 4.5em 1.5em 1.5em;
    border-radius: 0.8em;
	background:rgba(0,0,0,.7) url(./images/spinning.gif) no-repeat center 1.5em;
	background-size:auto 3em;
	color: white;
}





