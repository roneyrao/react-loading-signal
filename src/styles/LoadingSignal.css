.loading {
  top: 50%;
  left: 50%;
  z-index: 999;
  visibility: hidden;
  height: 0;
  text-align:center;
}
.loading > * {
  display: inline-block;
  transform: translate(-50%, -50%);
}
.masked {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 9999;
  display: flex;
  justify-content: center;
  align-items: center;
  transform: none;
}
.masked:before {
  content: '';
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(150, 150, 150, 0.4);
  position: absolute;
}
.masked.loading > * {
  transform: none;
  z-index: 1;
}

.local {
  composes: loading;
  position: absolute;
}
.localMasked {
  composes: local masked;
}

.global {
  composes: loading;
  position: fixed;
}
.globalMasked {
  composes: global masked;
}

.messageList {
  line-height: 1em;
  list-style: none;
  margin:0;
  padding:0;
}
.messageList > li {
  padding-top: 1em;
}
/*theme 'blobs'*/
.blobs {
  width: 12em;
  height: 1em;
  background: url(./images/blobs.gif) no-repeat center center;
  background-size: auto 100%;
}
/*theme 'spinning'*/
.spinning {
  min-width: 3em;
  max-width: 50%;
  padding: 4.5em 1.5em 1.5em;
  border-radius: 0.8em;
  background: rgba(0, 0, 0, 0.7) url(./images/spinning.gif) no-repeat center 1.5em;
  background-size: auto 3em;
  color: white;
  text-align: center;
  word-wrap: break-word;
}
/*theme 'progress'*/
.progress {
}
.progressBar {
  height: 5px;
  border: 1px solid #057927;
}
.progressBar > div{
  width: 0%;
  height: 100%;
  background: rgb(5, 121, 39);
}
