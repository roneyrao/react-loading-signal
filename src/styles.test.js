import styles from './styles';

test('emit json of className map from imported css', () => {
  expect(styles).toBeDefined();
  expect(styles.loading).toBeDefined();
});
test('className "composes" another', () => {
  expect(styles.local).toEqual(expect.stringContaining(styles.loading));
});
test('json full matches', () => {
  expect(styles).toMatchSnapshot();
});
