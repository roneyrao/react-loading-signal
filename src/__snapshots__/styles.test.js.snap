// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json full matches 1`] = `
Object {
  "blobs": "FlexLoading__blobs",
  "global": "FlexLoading__global FlexLoading__loading",
  "globalMasked": "FlexLoading__globalMasked FlexLoading__global FlexLoading__loading FlexLoading__masked",
  "loading": "FlexLoading__loading",
  "local": "FlexLoading__local FlexLoading__loading",
  "localMasked": "FlexLoading__localMasked FlexLoading__local FlexLoading__loading FlexLoading__masked",
  "masked": "FlexLoading__masked",
  "messageList": "FlexLoading__messageList",
  "spinning": "FlexLoading__spinning",
}
`;
